### DISCUSSION TITLE
What is `user_org_id` for?

### DISCUSSION THREAD

**Discussion by zc-devs** [woodpecker-ci/woodpecker#3360] [Ask a Question] [answered]
Keeping in mind [User-Org is Many-to-One relation](https://github.com/woodpecker-ci/woodpecker/pull/3319#discussion_r1481488743), I've made some testing.

In Gitea I created an Organization `gt` with a Team `gt-devs` and repository `gitea` as well as an Org `wp` with a team `wp-devs` and repository `woodpecker`. Then I added user `usr` to the both teams. I logged in Woodpecker by `usr` user and added `gitea` and `woodpecker` repositories.
Seems all works and user can be member of multiple organizations.

Looking in database:
```
woodpecker_test=# select user_id, user_login, user_org_id from users order by user_id;
 user_id |    user_login    | user_org_id
---------+------------------+-------------
       2 | usr              |           2
(5 rows)

woodpecker_test=# select * from orgs;
 id | name | is_user | private
----+------+---------+---------
  2 | usr  | t       | f
  6 | gt   | f       | f
  7 | wp   | f       | f
(7 rows)

```
I wonder, what is the purpose of `user_org_id` then?

**Comment by qwerty287** [comment, marked as answer]
Every user also has an org model in the `orgs` table to simplify features like user secrets.
This column links the corresponding org to the user.
In your case, user `usr` (ID 2) has the org with ID 2, thus `user_org_id` is 2 too.

**Reply by zc-devs** [reply]
So, there is two types of Orgs: `user` and `not user`.
Relation between User and UserOrg (flag is_user=true) is One-to-One, right?
Relation between User and OtherOrg (flag is_user=false) is Many-to-Many, right? And this relation is not direct, but via some other entity like `repository` (maybe)?

**Reply by qwerty287** [reply]
> Relation between User and UserOrg (flag is_user=true) is One-to-One, right?

Yes.

> Relation between User and OtherOrg (flag is_user=false) is Many-to-Many, right? And this relation is not direct, but via some other entity like repository (maybe)?

Yes, AFAIK this is not stored to db, but done in https://github.com/woodpecker-ci/woodpecker/blob/main/server/cache/membership.go.
